### DISCUSSION TITLE
Why doesn't the Advanced KSampler have 'denoise' ?

### DISCUSSION THREAD

**Discussion by drschwabe** [Comfy-Org/ComfyUI#1243] [General]
Changing _denoise_ value on normal KSampler node is handy cause you can easily get img2img by plugging in a pre-existing image into the latent_image property (and then throttling up or down the denoise; typically I find sweet spot is within 5 to 7.2 for enhancing/augmenting without disrupting scene too much) 

The Advanced node has latent_image property but no denoise .... why is that?   How can I get my pre-existing image to influence this node to acheive img2img like effect as we can do with normal KSampler node.

**Comment by ltdrdata** [comment]
`KSampler Advanced` is designed for low-level control compared to KSampler. You need to adjust `start_at_step` and `end_at_step` to implement it. The `denoise` parameter in `KSampler` simplifies this calculation.

Instead, `KSampler Advanced` controls the application of `denoise` through the steps at which `denoise` is applied. This allows for the separation of a single sampling process into multiple nodes. The separation of nodes also means that `conditioning`, `model`, `clip`  application, and other processes can be applied separately at each step.

And `KSampler Advance`d also enables applying forms of `denoise` that were not achievable with regular `KSampler`.

**Reply by drschwabe** [reply]
Thanks for clarifying.  Also found some more notes on the official docs: 

https://blenderneko.github.io/ComfyUI-docs/Core%20Nodes/Sampling/KSampler%20Advanced

Will have to play around with start_at_step and end_at_step to grasp how to get similar img2img results as ie- to acheive the equivalent KSampler basic denoising sweetspot of 5 to 7 for typical image enhancement purposes.

**Reply by ntdviet** [reply]
@drschwabe 
This is the very "basic" use of the advanced sampler
denoise = (steps - start_at_step) / steps

You can do advanced denoising by selecting the starting and ending steps mid-schedule to pick out the specific amount and weight of the noise you want to manipulate. For rough understanding, big noise means big stuff and change of image composition, small noise means finer stuff down to change of textures and surface. This is very model, sampler and schedule specific so you need to find out what step number changes what for your favorite setup.
When you are familiar with picking specific noise by sampler schedule, next you can do is injecting noise at specific steps.

**Reply by drschwabe** [reply]
goodstuff, thanks for sharing that.  So far with my experimentation with SDXL in a 2x Advanced KSampler setup (one initial and then sends to Refiner) is a start_step 15, end_at_step 25, with total 30 steps 
is about equivalent to that 5 - 7 range denoise 'sweet spot' on the simple KSampler.
